_posts/2020-05-18-mirror-kafka-eventhub.md

```diff
@@ -0,0 +1,219 @@
+        context.getLogger().info("Length:" + message.size());
+        message.forEach(singleMessage -> context.getLogger().info(singleMessage));
+    }
+}
+```
+
+Even in this case the `<eventhubs-namespace>` has to be replaced with the Event Hub namespace to connect to and the `EH_CONNECTION_STRING` environment variable has to be defined in the JSON application settings file containing the connection string as already used for the Kafka Mirror Maker configuration.
+
+For more information about Azure Functions connecting to Event Hub this [link](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-hubs-trigger) provides really useful information.
+The application can run even locally or be published on the Azure cloud.
+
+To try the entire pipeline, we just miss sending messages and see how they are mirrored to Event Hub triggering the above function to be executed.
+To do so, just use the `kafka-console-producer` command linme tool provided by every default Apache Kafka installation.
+Start a new pod in the Kubernetes cluster for hosting the producer and types a couple of JSON messages as follow.
+
+> They have to be JSON structured messages because the default Azure Functions deserializer expects this format
+
+```shell
+kubectl -n kafka run kafka-producer -ti --image=strimzi/kafka:0.17.0-kafka-2.4.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic testeh
+If you don't see a command prompt, try pressing enter.
+>{ "id": 1, "message": "Hello from Strimzi Mirror Maker" }
+>{ "id": 2, "message": "Here another mirrored message" }
+>
+```
+
+On the Azure Functions application, the messages will be logged as following.
+
+```shell
+[09/05/2020 14:02:17] Executing 'Functions.EventHubTriggerJava1' (Reason='', Id=6c54f1df-6a3d-4acc-a89a-4078b2049ddc)
+[09/05/2020 14:02:22] Java Event Hub trigger function executed.
+[09/05/2020 14:02:22] Length:1
+[09/05/2020 14:02:22] {"id":1,"message":"Hello from Strimzi Mirror Maker"}
+[09/05/2020 14:02:23] Function "EventHubTriggerJava1" (Id: 6c54f1df-6a3d-4acc-a89a-4078b2049ddc) invoked by Java Worker
+[09/05/2020 14:02:23] Executed 'Functions.EventHubTriggerJava1' (Succeeded, Id=6c54f1df-6a3d-4acc-a89a-4078b2049ddc)
+[09/05/2020 14:02:51] Executing 'Functions.EventHubTriggerJava1' (Reason='', Id=8edf21e3-94b3-4f4f-b2a7-8e92a311f9f0)
+[09/05/2020 14:02:51] Java Event Hub trigger function executed.
+[09/05/2020 14:02:51] Length:1
+[09/05/2020 14:02:51] {"id":2,"message":"Here another mirrored message"}
+[09/05/2020 14:02:51] Function "EventHubTriggerJava1" (Id: 8edf21e3-94b3-4f4f-b2a7-8e92a311f9f0) invoked by Java Worker
+[09/05/2020 14:02:51] Executed 'Functions.EventHubTriggerJava1' (Succeeded, Id=8edf21e3-94b3-4f4f-b2a7-8e92a311f9f0)
+```
+
+### Conclusion
+
+Integrating applications running on different platforms and different clouds is becoming quite common nowadays and the Hybrid Cloud based use cases are growing fast.
```

Comment:
The Hybrid Cloud here is not a specific product. So I think it should be in lower case

```suggestion
Integrating applications running on different platforms and different clouds is becoming quite common nowadays and the hybrid cloud based use cases are growing fast.
```